### PR TITLE
Fix versions-check cli command

### PR DESCRIPTION
**Pull request by grantila**
## Fix backstage-cli for certain yarn.lock situations

The `backstage-cli versions:check` (`--fix`) command fails when the package is part of a joint section in `yarn.lock`, e.g.:

`"@backstage/backend-common@npm:^0.21.3, @backstage/backend-common@npm:^0.21.6":`

with the error `Error: No entry data for @backstage/backend-common@^0.21.3`

The reason is that the code looks for a key in the `yarn.lock` object called `@backstage/backend-common@npm:^0.21.3` when in fact it's the full string `@backstage/backend-common@npm:^0.21.3, @backstage/backend-common@npm:^0.21.6`.

This PR fixes this, by looking up the full key using the `packages` object. It also fixes the `bump` command which uses `remove()` that has the same problem.

Also bumped `@yarnpkg/parsers` to stable `3.0.0` from `-rc.4` as it is released now.

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/devtools-backend/package.json
Comment (awanlin) [inline plugins/devtools-backend/package.json:44]: The DevTools plugin in this repo has been deprecated in favor of the version in the `community-plugins` repo. Given that we should probably not make this change.

**Review by awanlin — CHANGES_REQUESTED** [review]
Hi @grantila, just left a comment about the state of the DevTools plugin.

Will defer to the maintainers for final say though.

**Comment by grantila** [comment]
> Hi @grantila, just left a comment about the state of the DevTools plugin.
> 
> Will defer to the maintainers for final say though.

Makes perfect sense, I reverted that package upgrade

File: plugins/devtools-backend/package.json
Comment (benjdlambert) [reply plugins/devtools-backend/package.json]: @awanlin the devtools plugin hasn't been moved :)

**Review by awanlin — APPROVED** [review]
Approving for DevTools @grantila, the focus of this change looks good as well but would still like the maintainers to do a full review.

**Comment by benjdlambert** [comment]
Gonna bring in @Rugvip for a closer review here, as he wrote most of this already 👍

**Comment by grantila** [comment]
bump

**Review by Rugvip — APPROVED** [review]
Whaaat, was looking at this the other day and thought I had shipped it 😅 

Looks good, thank you! 👍 🎉

File: packages/cli-node/package.json
Comment (Rugvip) [reply packages/cli-node/package.json:38]: Oh, but with this bit too @grantila ^ 👀

**Comment by grantila** [comment]
Resolved by adding changeset and rebasing on master

**Comment by grantila** [comment]
Not sure why the tests for node 18 now fails, perhaps flaky and needs a re-run?

**Comment by awanlin** [comment]
Triggered a re-run, let's see how it goes

**Pull request merged**
